src/Select.js

```diff
@@ -250,30 +250,54 @@ let instanceId = 1;
 
 export default class Select extends Component<Props, State> {
   static defaultProps = defaultProps;
+  state = {
+    focusedOption: null,
+    focusedValue: null,
+    inputIsHidden: false,
+    isFocused: false,
+    menuOptions: { render: [], focusable: [] },
+    selectValue: [],
+  };
+
+  // Misc. Instance Properties
+  // ------------------------------
+
   blockOptionHover: boolean = false;
   clearFocusValueOnUpdate: boolean = false;
-  components: SelectComponents;
   commonProps: any; // TODO
-  controlRef: ElRef;
-  focusedOptionRef: ?HTMLElement;
+  components: SelectComponents;
   hasGroups: boolean = false;
-  input: ?ElRef;
-  inputIsHiddenAfterUpdate: ?boolean;
-  instancePrefix: string = '';
   initialTouchX: number = 0;
   initialTouchY: number = 0;
-  menuRef: ?ElRef;
+  inputIsHiddenAfterUpdate: ?boolean;
+  instancePrefix: string = '';
   openAfterFocus: boolean = false;
   scrollToFocusedOptionOnUpdate: boolean = false;
   userIsDragging: ?boolean;
-  state = {
-    focusedOption: null,
-    focusedValue: null,
-    inputIsHidden: false,
-    isFocused: false,
-    menuOptions: { render: [], focusable: [] },
-    selectValue: [],
+
+  // Refs
```

Comment:
I did a bunch of (somewhat unrelated) shuffling/cleanup on the Select class to make sense of things:

- `state` is now up with `defaultProps`
- named `ref` properties on the class are separated from miscellaneous properties
- the methods used to create the `refs` are now located with their applicable ref